### PR TITLE
Add management commands to restrict and throttle the api

### PR DESCRIPTION
**Pull request by stevenday**
This adds two basic management commands that interact with the redis database
that Varnish uses to perform api management.
- `api_keys_restrict_api` turns api key checking on and off (based on a setting in the django config)
- `api_keys_throttle_api` turns api throttling (rate limiting) on and off (again, based on settings in the django config)

### CODE REVIEW

File: api_keys/management/commands/api_keys_restrict_api.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+class Command(NoArgsCommand):
+    help = """Restrict access the api named in settings.REDIS_API_NAME if
+             settings.API_RESTRICT is True, un-restrict it otherwise."""
+
+    def handle_noargs(self, **options):
+        r = redis_connection()
+        if settings.API_RESTRICT:
+            r.set(RedisStrings.API_RESTRICT, '1')
+            if options['verbosity'] > 2:
```
Comment (dracos) [inline api_keys/management/commands/api_keys_restrict_api.py]: Probably want this to be > 1, not 2.

File: api_keys/utils.py [hunk 2]
```diff
@@ -16,3 +16,14 @@ def redis_connection():
         )
     return _connection
 
+class RedisStrings(object):
+    API_RESTRICT = 'api:{0}:restricted'.format(settings.REDIS_API_NAME)
+    API_THROTTLE = 'api:{0}:restricted'.format(settings.REDIS_API_NAME)
```
Comment (dracos) [inline api_keys/utils.py]: This has the same string as the key in the line above.

File: conf/general.yml-example [hunk 3]
```diff
@@ -48,9 +48,33 @@ SITE_BASE_URL: 'http://www.example.com'
...
+# Note: you can still set limits, or have no limit at all, for individual api
+# keys or IP addresses indepent of this setting
+API_THROTTLE_DEFAULT: 360
+
+# How long should users who go over the rate limit be blocked for? (Seconds)
+API_THROTTLE_BLOCK_TIME: 360
+
+# A list of api keys or IP addresses to exclude from rate limiting
+API_THROTTLE_UNLIMITED:
+    - '127.0.0.1'
```
Comment (dracos) [inline conf/general.yml-example]: This setting isn't used, it's using RATE_LIMIT if I've understood it right.

**Comment by dracos** [comment]
Typo in one commit, looks good at what it's doing (I guess it's the other end to see what any of this key setting does :) ).

File: conf/general.yml-example [hunk 4]
```diff
@@ -48,9 +48,33 @@ SITE_BASE_URL: 'http://www.example.com'
...
+# Should the API be throttled?
+API_THROTTLE: False
+
+# How long a time period should the api rate limiter counts hits over (seconds)
+API_THROTTLE_COUNTER_TIME: 360
+
+# How many hits during the API_RATE_LIMIT_COUNTER_TIME can a single user make
+# by default?
+# Note: you can still set limits, or have no limit at all, for individual api
+# keys or IP addresses indepent of this setting
```
Comment (dracos) [inline conf/general.yml-example]: Typo.

**Comment by stevenday** [comment]
Merged into master in 94711dd22c455365f35f2492f1e7df1614341466

**Pull request closed without merging**